### PR TITLE
TOOLS-2337 guinea pig in-repo Jenkinsfile with multibranch pipelines

### PR DESCRIPTION
**Pull request by cburroughs**

### CODE REVIEW

**Comment by trentm** [comment]
For @chudley's benefit. This represents in code in the repo what is encoded currently at https://jenkins.joyent.us/job/manta-deployment/configure    IOW it is beautiful.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    agent {
+        label '!platform:true && image_ver:15.4.1 && pkgsrc_arch:multiarch && pi:20151126T062538Z && jenkins_agent:2'
+    }
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '90'))
+        timestamps()
+    }
+
```
Comment (trentm) [inline Jenkinsfile]: @cburroughs https://jenkins.joyent.us/job/manta-deployment/configure includes `[x] Provide Node & npm bin/ folder to PATH` for sdcnode-v0.10.48-...     Should this Jenkinsfile (and perhaps others) do so as well?

Comment (trentm) [reply Jenkinsfile]: IIRC that sdcnode is there for `buildimage` to use. I wonder what it is picking up.

Comment (cburroughs) [reply Jenkinsfile]: In most (all?) cases where jobs that use sdcnode/engbld have that set, is a vestigial mountain-gorilla-ism.  Everything run through `make` should pick up sdcnode.  We do need a "put node on the path" for node libraries, see node-boray for an example.

**Pull request merged**